magento/magento2
# Exceptions when search product with sku like "42-"

### Description (*)
Try to search products with end of string like "-" or "-" 
The exception error appear `Unexpected $end` 
https://dev.mysql.com/doc/refman/8.0/en/fulltext-boolean.html

### Fixed Issues (if relevant)
1. magento/magento2#20716: Exceptions when search product with sku like "42-"
3. magento/magento2#9988 Quick search by SKU not working properly
### Manual testing scenarios (*)

1.Enable the MySql as search engine in configuration.
2.Search in the frontend with sku "42-" or "-"
3.No exception appear

### Contribution checklist (*)
